src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -219,11 +221,11 @@ public void Clear()
         /// <returns>true if item is found in the <see cref="ReadOnlyCollectionBuilder{T}"/>; otherwise, false.</returns>
         public bool Contains(T item)
         {
-            if ((Object)item == null)
+            if ((object)item == null)
```

Comment:
Yes. Typically JIT compiles separate methods for ref and val T. In either case the result of the condition is JIT-time constant. 
In fact it should be an input for the dead code elimination. Last time I checked JIT was smart enough to do that.